lib/tessel/access-point.js

```diff
@@ -0,0 +1,63 @@
+var Tessel = require('./tessel'),
+  commands = require('./commands'),
+  logs = require('../logs');
+
+function commitAndClose(self, resolve) {
+  return self.connection.exec(commands.commitWirelessCredentials())
+    .then(function(remoteProcess) {
+      remoteProcess.once('close', function() {
+        return self.connection.exec(commands.reconnectWifi())
+          .then(function(remoteProcess) {
+            return self.receive(remoteProcess).then(function() {
+              logs.info('Access Point credentials set!');
+
+              return self.connection.end()
+                .then(resolve);
+            });
+          });
+      });
+    });
+}
+
+Tessel.prototype.createAccessPoint = function(opts) {
+  var self = this;
+  var ssid = opts.ssid;
+  var password = opts.pass;
+  var security = opts.security;
+
+  return new Promise(function(resolve, reject) {
+    if (!ssid) {
+      return reject(new Error('Invalid credentials. Must set ssid'));
+    }
+
+    if (!password || !security) {
+      if (!security) {
```

Comment:
Shouldn't this be: 

```js
if (password || security) {
  if (security) {
```

Otherwise:

```
$ t2 ap --ssid=foo
INFO Looking for your Tessel...
INFO Connected to bishop over LAN
INFO Setting Access Point with SSID: foo and password: undefined and security mode: undefined
...
```

But I think you wanted it to display: 

```
$ t2 ap --ssid=foo
INFO Looking for your Tessel...
INFO Connected to bishop over USB
INFO Setting Access Point with SSID: foo
...
```

?